### DISCUSSION TITLE
CoreShop Messenger Monitor

### DISCUSSION THREAD

**Discussion by dpfaffenbauer** [pimcore/pimcore#13567] [Ideas]
I've been working on something that might be of interest to you. A Messenger Monitor to view pending and failed messages. (Failed only works if the transport has a failure_transport configured). It works with all transport receivers that support the `ListableReceiverInterface`). You can then also restart failed messages if the underlying error has been resolved. You can also delete them.

<img width="1357" alt="Screenshot 2022-11-12 at 15 04 09" src="https://user-images.githubusercontent.com/5981845/201477799-ee0ac97d-d722-4f6a-8920-baa9e57447bf.png">
<img width="1355" alt="Screenshot 2022-11-12 at 15 04 28" src="https://user-images.githubusercontent.com/5981845/201477805-b6c7cb23-8456-49e9-913a-074662da90e3.png">

So what do guys think of this? Is this something you need?

**Comment by fashxp** [comment]
Would be great to have something like that generic in the core. we even have an issue for that, see https://github.com/pimcore/pimcore/issues/10639

One challenge I see for that: what if the messaging queue is not doctrine based?

**Reply by dpfaffenbauer** [reply]
@fashxp as long as the receiver supports `ListableReceiverInterface`, we can get the Messages. As long as the Receiver supports `MessageCountAwareInterface`, we can get the count of Messages. 

Redis does not implement these interfaces, Amqp implements the `MessageCountAwareInterface`.

Alternatively, a different approach I followed is to duplicate the messages and store them somewhere else with the status. But I removed that again since it was getting to complicated to manage and Symfony doesn't always give you events for everything that is happening.

I would publish that as a CoreShop Extension.

**Reply by fashxp** [reply]
I see 👍 

CoreShop Extension won't help us in that case due to licensing topics, so at some point we would create something similar.

**Reply by dpfaffenbauer** [reply]
Why? How can you ever support 3rd-party extensions in enterprise then?

**Reply by fashxp** [reply]
only if it is dual licensed ... and that is only possible if the extension was developed against a Pimcore PCL instance.

**Comment by dpfaffenbauer** [comment]
For everybody who wanna give it a try: https://packagist.org/packages/coreshop/messenger-bundle
